tiledb/api/src/query/read/mod.rs

```diff
@@ -374,6 +384,262 @@ pub trait ReadQueryBuilder<'data>: QueryBuilder {
+        let mut agg_operation: *mut tiledb_channel_operation_t = out_ptr!();
+        let c_agg_name = agg_name.as_c_str().as_ptr();
+
+        if agg_type == AggregateType::Count {
+            context.capi_call(|ctx| unsafe {
+                ffi::tiledb_aggregate_count_get(
+                    ctx,
+                    core::ptr::addr_of_mut!(agg_operation)
+                        as *mut *const tiledb_channel_operation_t,
+                )
+            })?;
+        } else {
+            let c_attr_name: *const i8 =
+                attr_name.as_ref().unwrap().as_c_str().as_ptr();
+            match agg_type {
+                AggregateType::NullCount => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_null_count_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Sum => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_sum_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Max => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_max_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Min => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_min_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Mean => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_mean_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Count => unreachable!(),
+            };
+            context.capi_call(|ctx| unsafe {
+                ffi::tiledb_create_unary_aggregate(
+                    ctx,
+                    cquery,
+                    agg_operator,
+                    c_attr_name,
+                    &mut agg_operation,
+                )
+            })?;
+        }
+
+        // Apply aggregate to the default channel.
+        context.capi_call(|ctx| unsafe {
+            ffi::tiledb_channel_apply_aggregate(
+                ctx,
+                default_channel,
+                c_agg_name,
+                agg_operation,
+            )
+        })?;
+
+        Ok(AggregateBuilder::<Self, T> {
+            base: self,
+            agg_str: agg_name,
```

Comment:
Is the agg name used after the channel initialization? Is it needed to keep it around in the same way that the attribute name is?

Reply:
yes. Core TileDB stores the aggregate name in an unordered map:
/** Aggregates for the default channel, by output field name. */
  std::unordered_map<std::string, shared_ptr<IAggregator>>
      default_channel_aggregates_;